src/3rd_party-static/MessageBroker/src/lib_messagebroker/system.cpp

```diff
@@ -258,7 +267,58 @@ bool Mutex::Unlock() {
     return false;
   }
 
-  return ReleaseMutex(m_mutex);
+  return ReleaseMutex(m_mutex) != 0;
+}
+
+BinarySemaphore::BinarySemaphore() :
+  m_isUp(false) {
+  InitializeCriticalSection(&m_mutex);
+  InitializeConditionVariable(&m_cond);
+}
+
+BinarySemaphore::~BinarySemaphore() {
+  DeleteCriticalSection(&m_mutex);
+}
+
+void BinarySemaphore::Wait() {
+  // try to get exclusive access to the flag
+  EnterCriticalSection(&m_mutex);
```

Comment:
@AGaliuzov we're not performing any refactoring here. It's just merge of [sdl_core_winport](https://github.com/anosach-luxoft/sdl_core_winport)